File: JAVA/S9/L1/queue2.java
class Node{
    int data;
    Node next;
    Node(int data)
    {
        this.data=data;
        next=null;
    }

}
class que
{
    Node rear;
    Node front;
    que()
    {
        this.rear=null;
        this.front=null;
    }
}
class q{
    int isEmpty(que q)
    {
        if(q.rear==null || q.front==null)return 1;
        return 0;
    }
    void push(que q,int data)
    {
        Node newnode=new Node(data);
        if(q.rear==null)
        {
            // newnode.next=q.front;
            q.front=newnode;
            q.rear=newnode;
            return;
        }
       q.rear.next=newnode;
       q.rear=newnode;
    }
    void pop(que q)
    {
        if(isEmpty(q)==1)return ;
        q.front=q.front.next;


    }
    void display(que q)
    {
        Node temp=q.front;
        while(temp!=null)
        {
            System.out.print(temp.data+" ");
            temp=temp.next;

        }
    }
}
public class queue2
{
    public static void main(String[]args)
    {
        que q=new que();
        q qt=new q();
        qt.push(q,890);
        qt.push(q,809);
        qt.push(q,89);
        qt.push(q,89);
        qt.pop(q);
        qt.display(q);
    }
}

File: C/S1_BASIC/L1_BASIC/adt/queue/static/queue2.c
//using structure
#include<stdio.h>
#include<stdlib.h>
struct queue{
    int capacity;
    int rear;
    int front;
    int *arr;
};
struct queue* createQueue(int capacity){
    struct queue* q=(struct queue*)malloc(sizeof(struct queue));
    q->capacity=capacity;
    q->rear=0;
    q->front=0;
    q->arr=(int*)malloc(sizeof(int)*capacity);
    return 0;

}
int isFull(struct queue*q){
    return q->rear==q->capacity;
}
int isEmpty(struct queue *q){
    return q->front==q->rear;
}
void push(struct queue *q,int data){
    if(isFull(q)){
        printf("queue is full,can not insert data\n");
        return ;
    }
    q->arr[(q->rear)++]=data;
}
int pop(struct queue*q){
    if(isEmpty(q)){
        printf("queue is empty, can not remove element\n");
        return -1;
    }
    return q->arr[(q->front)++];
}
void display(struct queue *q){
    if(isEmpty(q)){
        printf("queue is empty,can not display element\n");
        return;
    }
    for(int i=q->front;i<=q->rear;i++){
        printf("%d ",q->arr[i]);
    }
}
int main(){
    struct queue *q=createQueue(5);
    push(q,56);
    display(q);
    return 0;

}